src/store/metadata/oxigraph/store.py

```diff
@@ -102,15 +112,95 @@ def get_editions(self, dataset_id: str) -> Optional[Dict]:  # pragma: no cover
         """
         Gets all editions of a specific dataset
         """
-        raise NotImplementedError
+        # Populate the graph from the database
+        graph = self.db
+
+        # Use the construct wrappers to pull the raw RDF triples
+        # (as one rdflib.Graph() for each function) and add them
+        # together to create a single Graph of the
+        # data we need.
+        result: Graph = construct_editions(graph, dataset_id)
+
+        # Serialize the graph into jsonld
+        data = json.loads(result.serialize(format="json-ld"))
+
+        # Use a context file to shape our jsonld, removing long form references
+        data = jsonld.flatten(
+            data, {"@context": constants.CONTEXT, "@type": "hydra:Collection"}
+        )
+        editions_graph = _get_single_graph_for_field(data, "@type")
+
+        # TODO Fix context weirdness - at the moment, the flatten() method is changing @type to `versions_url` and `editions` to `versions`
+        editions_graph["@type"] = "hydra:Collection"
+        editions_graph["editions"] = editions_graph.pop("versions")
+
+        for idx, edition in enumerate(editions_graph["editions"]):
```

Comment:
you can use a comprehension for this

```
        editions_graph["editions"] = [
            self.get_edition(dataset_id, x["identifier"]) for x in editions_graph["editions"]
        ]
```

unless there's multiple nested conditions. you can almost always use a comprehension, handy string to have on your bow.

as a rule of thumb in python, use a comprehension until such time as it impacts readability, at which point (for the sake of the next developers sanity) you'll want to iterate.